java-openjdk/riscv64.patch

```diff
@@ -1,23 +1,23 @@
-Index: PKGBUILD
-===================================================================
---- PKGBUILD	(revision 444712)
-+++ PKGBUILD	(working copy)
-@@ -26,15 +26,22 @@
+--- PKGBUILD	2022-09-28 16:19:15.520937763 +0800
++++ PKGBUILD	2022-09-28 22:46:50.618989064 +0800
+@@ -26,15 +26,24 @@
  source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
          freedesktop-java.desktop
          freedesktop-jconsole.desktop
 -        freedesktop-jshell.desktop)
 +        freedesktop-jshell.desktop
 +
 +        # see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-18-openjdk
++        config.guess
```

Comment:
This config.guess is the same as config.guess provided by automake, you should not use vendor one.

use `cp -v /usr/share/automake-1.16/config.guess .`

It's completely the same as [file on master](https://git.savannah.gnu.org/cgit/automake.git/tree/lib/config.guess), and only changed unrelated content from the latest stable.

```diff
--- /usr/share/automake-1.16/config.guess	2021-10-05 03:45:02.000000000 +0800
+++ config.guess	2022-09-28 23:02:20.243704826 +0800
@@ -1,14 +1,14 @@
 #! /bin/sh
 # Attempt to guess a canonical system name.
-#   Copyright 1992-2021 Free Software Foundation, Inc.
+#   Copyright 1992-2022 Free Software Foundation, Inc.
 
 # shellcheck disable=SC2006,SC2268 # see below for rationale
 
-timestamp='2021-06-03'
+timestamp='2022-01-09'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
-# the Free Software Foundation; either version 3 of the License, or
+# the Free Software Foundation, either version 3 of the License, or
 # (at your option) any later version.
 #
 # This program is distributed in the hope that it will be useful, but
@@ -60,7 +60,7 @@
 GNU config.guess ($timestamp)
 
 Originally written by Per Bothner.
-Copyright 1992-2021 Free Software Foundation, Inc.
+Copyright 1992-2022 Free Software Foundation, Inc.
 
 This is free software; see the source for copying conditions.  There is NO
 warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE."
@@ -437,7 +437,7 @@
 	# This test works for both compilers.
 	if test "$CC_FOR_BUILD" != no_compiler_found; then
 	    if (echo '#ifdef __amd64'; echo IS_64BIT_ARCH; echo '#endif') | \
-		(CCOPTS="" $CC_FOR_BUILD -E - 2>/dev/null) | \
+		(CCOPTS="" $CC_FOR_BUILD -m64 -E - 2>/dev/null) | \
 		grep IS_64BIT_ARCH >/dev/null
 	    then
 		SUN_ARCH=x86_64
@@ -929,6 +929,9 @@
     i*:PW*:*)
 	GUESS=$UNAME_MACHINE-pc-pw32
 	;;
+    *:SerenityOS:*:*)
+        GUESS=$UNAME_MACHINE-pc-serenity
+        ;;
     *:Interix*:*)
 	case $UNAME_MACHINE in
 	    x86)
@@ -1522,6 +1525,9 @@
     i*86:rdos:*:*)
 	GUESS=$UNAME_MACHINE-pc-rdos
 	;;
+    i*86:Fiwix:*:*)
+	GUESS=$UNAME_MACHINE-pc-fiwix
+	;;
     *:AROS:*:*)
 	GUESS=$UNAME_MACHINE-unknown-aros
 	;;

```